src/main/java/org/apache/maven/plugins/scripting/AbstractScriptEvaluator.java

```diff
@@ -32,13 +35,17 @@ abstract class AbstractScriptEvaluator {
 
     /**
      * @param bindings not null bindings to provide to the script to execute
+     * @param log engine logger if context aware.
      * @return the possibly null result the script produced
      * @throws UnsupportedScriptEngineException when the engine is not configured correctly
      * @throws ScriptException  if an error occurs in script.
      */
-    public final Object eval(Bindings bindings) throws ScriptException, UnsupportedScriptEngineException {
+    public final Object eval(Bindings bindings, Log log) throws ScriptException, UnsupportedScriptEngineException {
```

Comment:
This changes existing public API. Consider adding a new method instead, especially since this argument isn't used on all paths.

Reply:
It is not really public as the class is not public. I have an idea how to get rid of this (https://github.com/apache/maven-scripting-plugin/pull/26) but since the Log could be passed in without loosing its type, I would prefer this (non-public!) change.

Reply:
OK. In that case can this method be non-public? Or is there a missing `@Override` annotation to add here?

Reply:
I changed it to protected. So are the other methods already. WDYT?

Reply:
Guess we don't care much, there is no real public API, all are internals of the mojo so private or public are likely equivalent to be extreme in the way to say it.